source/uwp/Renderer/lib/Util.cpp

```diff
@@ -161,677 +132,431 @@ try
     {
-        HString url;
-        RETURN_IF_FAILED(backgroundImage->get_Url(url.GetAddressOf()));
-        *isValid = url.IsValid();
+        return !backgroundImage.Url().empty();
     }
-    return S_OK;
+    return false;
 }
 
-void GetUrlFromString(_In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveHostConfig* hostConfig,
-                      _In_ HSTRING urlString,
-                      _Outptr_ ABI::Windows::Foundation::IUriRuntimeClass** url)
+winrt::Uri GetUrlFromString(winrt::AdaptiveHostConfig const& hostConfig, winrt::hstring const& urlString)
 {
-    ComPtr<ABI::Windows::Foundation::IUriRuntimeClassFactory> uriActivationFactory;
-    THROW_IF_FAILED(GetActivationFactory(HStringReference(RuntimeClass_Windows_Foundation_Uri).Get(), &uriActivationFactory));
-
-    ComPtr<ABI::Windows::Foundation::IUriRuntimeClass> localUrl;
+    winrt::Uri uri{nullptr};
 
-    // Try to treat URI as absolute
-    boolean isUrlRelative = FAILED(uriActivationFactory->CreateUri(urlString, localUrl.GetAddressOf()));
-
-    // Otherwise, try to treat URI as relative
-    if (isUrlRelative)
+    if (const auto uriFromAbsoluteUri = winrt::Uri{urlString})
+    {
+        return uriFromAbsoluteUri;
+    }
+    else
     {
-        HString imageBaseUrl;
-        THROW_IF_FAILED(hostConfig->get_ImageBaseUrl(imageBaseUrl.GetAddressOf()));
+        winrt::hstring imageBaseUrl = hostConfig.ImageBaseUrl();
 
-        if (imageBaseUrl.Get() != nullptr)
+        if (const auto uriFromRelativeUri = winrt::Uri{imageBaseUrl, urlString})
         {
-            THROW_IF_FAILED(uriActivationFactory->CreateWithRelativeUri(imageBaseUrl.Get(), urlString, localUrl.GetAddressOf()));
+            return uriFromRelativeUri;
         }
     }
-
-    THROW_IF_FAILED(localUrl.CopyTo(url));
+    return nullptr;
 }
 
-
-Color GenerateLHoverColor(const Color& originalColor)
+winrt::Windows::UI::Color GenerateLHoverColor(winrt::Windows::UI::Color const& originalColor)
 {
     const double hoverIncrement = 0.25;
 
-    Color hoverColor;
+    winrt::Windows::UI::Color hoverColor;
     hoverColor.A = originalColor.A;
-    hoverColor.R = originalColor.R - static_cast<BYTE>(originalColor.R * hoverIncrement);
-    hoverColor.G = originalColor.G - static_cast<BYTE>(originalColor.G * hoverIncrement);
-    hoverColor.B = originalColor.B - static_cast<BYTE>(originalColor.B * hoverIncrement);
+    hoverColor.R = originalColor.R - static_cast<uint8_t>(originalColor.R * hoverIncrement);
+    hoverColor.G = originalColor.G - static_cast<uint8_t>(originalColor.G * hoverIncrement);
+    hoverColor.B = originalColor.B - static_cast<uint8_t>(originalColor.B * hoverIncrement);
     return hoverColor;
 }
 
-DateTime GetDateTime(unsigned int year, unsigned int month, unsigned int day)
+winrt::DateTime GetDateTime(unsigned int year, unsigned int month, unsigned int day)
 {
+    // TODO: investigate the midnight bug. If the timezone will be ahead of UTC we can do -1 day when converting
```

Comment:
@paulcam206 could you please have a look at this?

Reply:
I'd file a bug to revisit. We'll want to get changes here under test, etc.

Reply:
https://github.com/microsoft/AdaptiveCards/issues/6900